enclave-runtime/src/lib.rs

```diff
@@ -805,10 +813,18 @@ where
 ///
 /// Todo: This will probably be used again if we decide to make sidechain optional?
 #[allow(unused)]
```

Comment:
I know that's not from this PR, but it still disturbs me - why are we not using this function? Respective, how are we iterating over shards if not in this function? Because I don't think the comment from above `This will probably be used again if we decide to make sidechain optional?` applies. Sidechain and top pool are coupled, so we will only use this function if we have the sidechain included..?

Reply:
Yes the comment is misleading. 
1. It is in fact used, but only for tests
2. Iterating over the shards is done in the slot worker, see [here](https://github.com/integritee-network/worker/blob/b1f1bbd45ed16bf490e3393878dedf97a0083928/sidechain/consensus/slots/src/lib.rs#L262-L294)
3. If the sidechain is optional and disabled, this still won't be used, since the top-pool is coupled to the sidechain. So you're absolutely right 👍 This function will be refactored and the comment removed in 2 PRs after this one 😄